influxdb3_wal/src/snapshot_tracker.rs

```diff
@@ -118,6 +138,29 @@ impl SnapshotTracker {
         })
     }
 
+    fn snapshot_all(&mut self) -> Option<SnapshotInfo> {
+        let n_periods_to_take = self.wal_periods.len() - 1;
+        let wal_periods: Vec<WalPeriod> = self.wal_periods.drain(0..n_periods_to_take).collect();
+        let max_time = wal_periods
+            .iter()
+            .map(|period| period.max_time)
+            .max()
+            .unwrap();
+        let t = max_time - (max_time.get() % self.gen1_duration.as_nanos())
+            + self.gen1_duration.as_nanos();
+        let last_wal_sequence_number = wal_periods.last().unwrap().wal_file_number;
+        let snapshot_details = SnapshotDetails {
+            snapshot_sequence_number: self.increment_snapshot_sequence_number(),
+            end_time_marker: t.get(),
+            last_wal_sequence_number,
+        };
+
+        Some(SnapshotInfo {
+            snapshot_details,
+            wal_periods,
+        })
+    }
```

Comment:
```suggestion
    fn snapshot_all(&mut self) -> Option<SnapshotInfo> {
        let n_periods_to_take = self.wal_periods.len().checked_sub(1)?;
        let wal_periods: Vec<WalPeriod> = self.wal_periods.drain(0..n_periods_to_take).collect();
        let max_time = wal_periods
            .iter()
            .map(|period| period.max_time)
            .max()?;
        // TODO - use checked to prevent overflows here(?):
        let t = max_time - (max_time.get() % self.gen1_duration.as_nanos())
            + self.gen1_duration.as_nanos();
        let last_wal_sequence_number = wal_periods.last()?.wal_file_number;
        let snapshot_details = SnapshotDetails {
            snapshot_sequence_number: self.increment_snapshot_sequence_number(),
            end_time_marker: t.get(),
            last_wal_sequence_number,
        };

        Some(SnapshotInfo {
            snapshot_details,
            wal_periods,
        })
    }
```
Either this function doesn't need to return an `Option` (and panics) or it could be made safer (which is what my suggestion aims at).